File: Core/Entities/Product.cs
using System;

namespace Core.Entities;

public class Product : BaseEntity
{
    // public int Id { get; set; } //It does not need anymore cause ID already inharited from the base entity
    public required string Name { get; set; }
    public required string Description { get; set; }
    public  decimal Price { get; set; }
    public required string PictureUrl { get; set; }
    public required string Type { get; set; }
    public required string Brand { get; set; }
    public int QuantityInStock { get; set; }
}

File: Client/src/app/shared/models/product.ts

// export interface Product  {

// }
// we will use type instead of an interface 


//camelCasig
 export type Product  ={
      id : number;
      name :string;
      description : string;
      price : number; // we don't have decimal for typeScipt
      pictureUrl : string;
      type : string;
      brand : string;
      quantityInStock : number;
 }